src/mi.rs

```diff
@@ -23,33 +34,36 @@ where
 /// random variables.
 pub struct SpecificCorrelation {
     normalize: bool,
+    smoothing: Smoothing,
 }
 
 impl SpecificCorrelation {
     /// Construct a new specific correlation function.
     ///
     /// If normalization is enable, the result will lie between -1 and 1.
-    pub fn new(normalize: bool) -> Self {
+    pub fn new(normalize: bool, smoothing: Smoothing) -> Self {
         SpecificCorrelation {
-            normalize: normalize,
+            normalize,
+            smoothing
         }
     }
 }
 
 impl<V> MutualInformation<V> for SpecificCorrelation
-where
-    V: Eq + Hash,
+    where
+        V: Eq + Hash,
 {
     fn mutual_information(
```

Comment:
Oh, one addition: one nicety of separating these is that most of the arguments that are passed around for `sc` could be removed, I think it would just need the tuple and a reference to the smoothing implementation.

Reply:
Thanks for the suggestion! I wasn't too happy about fiddling the smoothing into existing methods either and this will solve the issue 👍